quix-frontend/client/src/lib/sql-autocomplete/adapter/sql-autocpmplete-adapter-utills.ts

```diff
@@ -0,0 +1,304 @@
+import { TableInfo } from "../sql-context-evaluator";
+
+interface ObjectChild {
+  name: string;
+  dataType: any;
+}
+
+export function getObjectChildren(obj: Record<string, any>, parentName = ''): ObjectChild[] {
+  const children: ObjectChild[] = [];
+
+  for (const key in obj) {
+    if (typeof obj[key] === 'object' && obj[key] !== null) {
+      const childName = parentName ? `${parentName}.${key}` : key;
+      children.push({ name: childName, dataType: obj[key] });
+      children.push(...getObjectChildren(obj[key], childName));
+    } else {
+      const childName = parentName ? `${parentName}.${key}` : key;
+      children.push({ name: childName, dataType: obj[key] });
+    }
+  }
+
+  return children.map(child => {
+    child.name = child.name.replace(/\.dataType$/, '');
+    return child;
+  });
+}
+
+const enum StartingOptions {
+  row = "row(",
+  map = "map(",
+  array = "array(",
```

Comment:
can you explain the difference between map and array?

Reply:
i dont fully understand your comment

Reply:
what is the difference between map and array in Trino types

Reply:
no real diffrence for me, but 
i write the type on the side so it helps me determine the type.